MoreCyclopsUpgrades/Modules/Recharging/Nuclear/DepletedNuclearModule.cs

```diff
@@ -0,0 +1,55 @@
+﻿using SMLHelper;
+using SMLHelper.Patchers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using UnityEngine;
+
+namespace MoreCyclopsUpgrades
+{
+    public class DepletedNuclearModule
+    {
+        public static TechType DepletedCYNUKM { get; protected set; }
+
+        public static void Patch()
+        {
+            DepletedCYNUKM = TechTypePatcher.AddTechType("DepletedCyclopsNuclearModule", "Depleted Cyclops Nuclear Module", "A depleted Cyclops Nuclear Module");
+            TechType dummy = TechTypePatcher.AddTechType("CyclopsNuclearModuleRefil", "Reload Cyclops Nuclear Module", "Reload a Depleted Cyclops Nuclear Module with a Reactor Rod");
+            Atlas.Sprite sprite = SpriteManager.Get(TechType.DepletedReactorRod);
+            CustomSprite customSprite = new CustomSprite(dummy, sprite);
+            CustomCraftNode customCraftNode = new CustomCraftNode(dummy, CraftTree.Type.Workbench, "CyclopsMenu/DepletedCyclopsNuclearModule");
+            TechDataHelper techData = new TechDataHelper()
+            {
+                _craftAmount = 0,
+                _ingredients = new List<IngredientHelper>()
+                {
+                    new IngredientHelper(DepletedCYNUKM, 1),
+                    new IngredientHelper(TechType.ReactorRod, 1)
+                },
+                _linkedItems = new List<TechType>()
+                {
+                    NuclearCharger.CyNukBatteryType,
+                    TechType.DepletedReactorRod
+                },
+                _techType = dummy
+            };
+
+            CustomSpriteHandler.customSprites.Add(customSprite);
+            CraftDataPatcher.customTechData.Add(dummy, techData);
+            CraftTreePatcher.customNodes.Add(customCraftNode);
+
+            CustomPrefabHandler.customPrefabs.Add(new CustomPrefab("DepletedCyclopsNuclearModule", "WorldEntities/Natural/DepletedCyclopsNuclearModule", dummy, DepletedGameObject));
+        }
+
+        public static GameObject DepletedGameObject()
+        {
+            GameObject prefab = Resources.Load<GameObject>("WorldEntities/Natural/DepletedReactorRod");
+            GameObject obj = UnityEngine.Object.Instantiate(prefab);
+
+            obj.GetComponent<PrefabIdentifier>().ClassId = "DepletedCyclopsNuclearModule";
```

Comment:
Missing update to `<TechTag>().Type` component.
Look to any of the other prefab methods in this project for examples.

Reply:
`DepletedReactorRod` does not have a `TechTag` component

Reply:
I'll want to double-check to be sure. Cause if not, this will need to use a different base prefab instead, one that has a TechType tag that the code in the recharge loop can check for.

Reply:
Or..., we could do `AddComponent <TechTag>().Type`

Reply:
I suppose there's no reason we couldn't do that if this PreFab doesn't have it already.
If that's the route you want to go, I would also make sure that the `SpawnDepletedModule()` method sets the `Type` to the correct `TechType` when the item is first created.

I'd have to test that out to be sure if it's necessary, but it certainly won't hurt and it will give us the confidence to know that it's there and that it can be referenced by anything that needs to.

Reply:
The SpawnDepletedModule() method gets the prefab made by SMLHelper, in which the type is set, so IMHO, I don't think it is required to set it again, but nothing bad can happen if we set it again

Reply:
Looking at the current diff, that was true of the old `SpawnDepletedRod()` method, but the new `SpawnDepletedModule()` method uses `GameObject prefab = Resources.Load<GameObject>("WorldEntities/Natural/DepletedCyclopsNuclearModule");`

See diffs on `NuclearChargingManager.cs`

Reply:
Yes, I know. That's the part that i'm concerned about. SMLHelper makes the prefabs last, so < now that I think about it, it wont work 100% > they might not be available for use when that code is called

Reply:
Don't merge yet